src/main/sensors/compass.c

```diff
@@ -462,35 +460,44 @@ uint32_t compassUpdate(timeUs_t currentTimeUs)
 
     flightDynamicsTrims_t *magZero = &compassConfigMutable()->magZero;
 
-    if (tCal != 0) {
-        if (cmpTimeUs(tCal, currentTimeUs) > 0) {
-            LED0_TOGGLE;
-
+    timeDelta_t magCalTimeRemaining = 0;
```

Comment:
Only fact that calibration is in progress is important. And `mapLambdaGain` if valid only during actual callibration, maybe record it only then.

```suggestion
    bool recordMapLambdaGain = false;
```

Reply:
I agree.  I used a counting down number while debugging to check that the timing was correct; it was nice to see the countdown timer start at 15 on initiation, step down towards 0, and then the kick up to 30s once the threshold was reached. I needed this because there was a Configurator display bug where the calibrate button didn't seem to match the duration of the cal process.

But now, I agree, we only need a true/false indicator of whether there is a cal in progress.

I have simplified these things a bit more thanks to your feedback.